### DISCUSSION TITLE
I have uploaded my model-viewer ar application to amplify and it doesn't work exclusively on android

### DISCUSSION THREAD

**Discussion by ViniTamanhao** [google/model-viewer#4277] [Q&A]
I have been working on a restaurant application that shows the menu with different 3d observation modes for some certain dishes. I have used React along with AWS's s3 for model storage. After testing it locally several times, both with android and iOS, i have decided to deploy my web application with Amplify, for the sake of simplicity. Everything was normal until I decided to test the amplify application on android. The whole screen was normal, as it was supposed to be, and the ar button shows as intended. But when clicked, it does nothing. No error message, no sound, no refresh. I have already checked if the 3d model link is correct and it is. Is there any Amplify config I am sort of overlooking here?

this is my code
<img width="511" alt="Screenshot 2023-05-10 at 19 13 48" src="https://github.com/google/model-viewer/assets/105015723/ae032c6a-0596-4282-9f62-112452807edf">

**Comment by elalish** [comment]
First, there should be no need for an AR if statement - we'll already hide the AR button if the device/browser doesn't support it. I don't know much about amplify - is this a website or an app? If you can access the JS console, it should print some information when the AR button is clicked. Can you paste that in here?

**Reply by ViniTamanhao** [reply]
I'll pass you the link

https://www.bytenfood.com/secao/000f0524-8691-4ecf-ab4f-fd18bdaa558f/itens

**Reply by ViniTamanhao** [reply]
in regards to the if statement, it's just a double check in case we have only one of the models... we have made some tests and it was showing up the button even if only the android model was available, and since we are only working with glb and usdz, not glft, such a validation was necessary

**Reply by ViniTamanhao** [reply]
> First, there should be no need for an AR if statement - we'll already hide the AR button if the device/browser doesn't support it. I don't know much about amplify - is this a website or an app? If you can access the JS console, it should print some information when the AR button is clicked. Can you paste that in here?

the hardest part about is that i can only debug on a android device for such a problem... and I just have a iPhone. Currently my boss is doing the debugging for me and sending the errors... but from what I've seen it seems as if the application just ignored the clicking... as crazy as it sounds hahahhaha

**Reply by ViniTamanhao** [reply]
> I'll pass you the link
> 
> https://www.bytenfood.com/secao/000f0524-8691-4ecf-ab4f-fd18bdaa558f/itens

feel free to inspect and see the usage of the model-viewer component!

**Reply by elalish** [reply]
I'd recommend picking up a cheap android phone - it's basically impossible to debug AR otherwise. I saw some CORS errors when I tried to load your GLB, so check your server config.

**Reply by ViniTamanhao** [reply]
> I'd recommend picking up a cheap android phone - it's basically impossible to debug AR otherwise. I saw some CORS errors when I tried to load your GLB, so check your server config.

it was a CORS error... I forgot a little simple configuration on my S3 bucket... regardless, thanks for your help broda, fr. Have a great day!
